src/intrprtr.c

```diff
@@ -277,28 +233,23 @@ static void FinishAndCallFakeFuncExpr(Obj stackNams)
 **  return-void-statement was interpreted. If 'IntrEnd' returns 'STATUS_QUIT',
 **  then a quit-statement was interpreted.
 */
-void IntrBegin ( Obj frame )
+void IntrBegin(IntrState * intr,  Obj frame)
 {
-    /* remember old interpreter state                                      */
-    if (!STATE(IntrState))
-        STATE(IntrState) = NEW_PLIST(T_PLIST, 16);
-    PushPlist(STATE(IntrState), STATE(StackObj));
```

Comment:
Real change: we don't need to push `StackObj` onto a stack here and restore it later, as the interpreter state contains `StackObj` and is itself allocated on the C calling stack.

Reply:
Note that we did not save IntrIgnoring, IntrCoding, IntrReturning here, but relied on the caller to do so.